tty/uart16550/uart16550.c

```diff
@@ -187,6 +189,10 @@ static void uart_ioctl(unsigned int port, msg_t *msg)
 	if (uart == NULL) {
 		err = -EINVAL;
 	}
+	else if ((uart == &uart_common.uarts[UART16550_CONSOLE_USER]) && (req == KIOEN)) {
```

Comment:
`uart == &uart_common.uarts[UART16550_CONSOLE_USER]` condition should be verified inside if block to avoid the risk of passing a KIOEN request to `libtty_ioctl`.

Reply:
what's bad about it?

Reply:
`KIOEN` is not a TTY ioctl and could potentially be misinterpreted as a different ioctl.